rubyUtils.gradle

```diff
@@ -274,13 +286,37 @@ tasks.register("downloadAndInstallJRuby", Copy) {
     }
     exclude "**/stdlib/rdoc/**"
     exclude "**/did_you_mean-*/evaluation/**" // licensing issue https://github.com/jruby/jruby/issues/6471
+    exclude "**/lib/jni/**/**"
 
     includeEmptyDirs = false
     into "${projectDir}/vendor/jruby"
 }
 
 downloadAndInstallJRuby.onlyIf { customJRubyDir == "" }
 
+tasks.register("downloadAndInstallPreviousJRubyFFI", Copy) {
+    dependsOn=[downloadPreviousJRuby]
+    description "Install previous JRuby FFI files in the new JRuby directory"
+    inputs.file(previousJrubyTarPath)
+    outputs.dir("${projectDir}/vendor/jruby/lib/jni")
+    outputs.file("${projectDir}/vendor/jruby/lib/jruby.jar")
+    from tarTree(downloadPreviousJRuby.dest)
+    eachFile { f ->
+        f.path = f.path.replaceFirst("^jruby-${previousJRubyVersion}", '')
+    }
+    // Copy only the previous builds of libjffi that changed from JRuby 9.3.9.0 to 9.3.10.0
+    include "**/lib/jni/**/*"
+
+    includeEmptyDirs = false
+    into "${projectDir}/vendor/jruby/tmp"
+    doLast {
+        ant.jar(destfile: "${projectDir}/vendor/jruby/lib/jruby.jar", update: true) {
+            fileset(dir: "${projectDir}/vendor/jruby/tmp/lib")
+        }
+        delete files("${projectDir}/vendor/jruby/tmp")
+    }
+}
```

Comment:
NIT: minor re-arrange, return to only getting the FFI libs that are known to have changed, and output information about what we are injecting. I'm ok with filing this as a follow-up.

```suggestion
tasks.register("downloadAndInstallPreviousJRubyFFI", Copy) {
    dependsOn=[downloadPreviousJRuby]
    description "Install previous JRuby FFI files in the new JRuby directory"

    inputs.file(previousJrubyTarPath)
    outputs.dir("${projectDir}/vendor/jruby/lib/jni")
    outputs.file("${projectDir}/vendor/jruby/lib/jruby.jar")

    includeEmptyDirs = false

    from(tarTree(downloadPreviousJRuby.dest)) {
        // Copy only the previous builds of libjffi that changed from JRuby 9.3.9.0 to 9.3.10.0
        include "**/lib/jni/aarch64-Linux/libjffi-1.2.so"
        include "**/lib/jni/arm-Linux/libjffi-1.2.so"
        include "**/lib/jni/i386-Linux/libjffi-1.2.so"
        include "**/lib/jni/x86_64-Linux/libjffi-1.2.so"

        eachFile { f ->
            f.path = f.path.replaceFirst("^jruby-${previousJRubyVersion}", '')
            println "Installing Previous FFI: ${f.path}"
        }
    }
    into "${projectDir}/vendor/jruby/tmp"

    doLast {
        ant.jar(destfile: "${projectDir}/vendor/jruby/lib/jruby.jar", update: true) {
            fileset(dir: "${projectDir}/vendor/jruby/tmp/lib")
        }
        delete files("${projectDir}/vendor/jruby/tmp")
    }
}
```